Library/Homebrew/formulary.rb

```diff
@@ -489,17 +491,17 @@ class FromUrlLoader < FormulaLoader
     def initialize(url, from: nil)
       @url = url
       @from = from
-      uri = URI(url)
-      formula = File.basename(uri.path, ".rb")
-      super formula, HOMEBREW_CACHE_FORMULA/File.basename(uri.path)
+      uri_path = T.must(URI(url).path)
```

Comment:
I think this should just be an `ArgumentError` instead. The `path` is only guaranteed to be present since we only pass specific URLs to this.

Reply:
Could I trouble you to be more specific in how you would like this to look? I'm not following where you want to raise an `ArgumentError`. (The `T.must` will raise a `TypeError` here on a falsy `path`, which I think is the correct error type, but also not something worth splitting hairs over.)

Reply:
Something like

```ruby
raise ArgumentError, "URL has no path component" unless uri.path
```

Reply:
We'll probably never hit this, but semantically an argument error makes more sense to me here.

Reply:
sorbet (rightfully) doesn't assume successive invocations of the same function will return the same type. so this would actually need to be something like:
```ruby
uri_path = uri.path
raise ArgumentError, "URL has no path component" unless uri.path
```
which is a lot of code just to return a specific error type, even if I agreed with it. (Which I'm still not sold on – if this is not a `TypeError`, I'm not sure what would be.)

Reply:
Semantically, it's an argument error caused by invalid input (i.e. a URL without path component), which in turn leads to a type error if left unhandled.